OMCompiler/SimulationRuntime/c/simulation/solver/kinsolSolver.c

```diff
@@ -62,202 +62,285 @@
 #include <stdlib.h>
 #include <math.h>
 
-#include <kinsol/kinsol.h>
-#include <kinsol/kinsol_impl.h>
-#include <kinsol/kinsol_dense.h>
-#include <kinsol/kinsol_lapack.h>
-#include <kinsol/kinsol_direct.h>
-#include <kinsol/kinsol_klu.h>
-#include <sundials/sundials_nvector.h>
-#include <nvector/nvector_serial.h>
-#include <sundials/sundials_types.h>
-#include <sundials/sundials_math.h>
 
+/* Function prototypes */
+static int nlsKinsolResiduals(N_Vector x, N_Vector f, void *userData);
+static void nlsKinsolErrorPrint(int errorCode, const char *module, const char *function,
+                                char *msg, void *userData);
+static void nlsKinsolInfoPrint(const char *module, const char *function, char *msg, void *user_data);
+static int nlsSparseJac(N_Vector vecX, N_Vector vecFX, SUNMatrix Jac,
+                        void *userData, N_Vector tmp1, N_Vector tmp2);
+int nlsSparseSymJac(N_Vector vecX, N_Vector vecFX, SUNMatrix Jac, void *userData,
+                    N_Vector tmp1, N_Vector tmp2);
+static int nlsDenseJac(long int N, N_Vector vecX, N_Vector vecFX, SUNMatrix Jac,
+                       void *userData, N_Vector tmp1, N_Vector tmp2);
+static void nlsKinsolJacSumSparse(SUNMatrix A);
+static void nlsKinsolJacSumDense(SUNMatrix A);
+
+
+/**
+ * @brief Checks given KINSOL flag and reports potential error.
+ *
+ * @param flag          Return value of Kinsol routine.
+ * @param functionName  Name of Kinsol function that returned the flag.
+ */
+void checkReturnFlag_KIN(int flag, const char* functionName) {
+
+  switch(flag) {
+    case KIN_SUCCESS:
+    case KIN_INITIAL_GUESS_OK:
```

Comment:
I would like to make a general comment on the coding style of scopes/function bodies in the c runtime:

### Option 1
    void foo() {
     
        [...]
    }

### Option 2
    void foo() {
       [...]
    }

### Option 3
    void foo()
    {
        [...]
    }

This is how often we use the three patterns in the runtime:

    \) \{\n\n  ->   23 hits (e.g. in line 87)
    \) \{\n    -> 2418 hits (e.g. in line 89)
    \)\n\{\n   -> 1563 hits  

So what is my point? It would be good to keep the coding consistent. It seems that option 1 isn't really used at all. Option 2 is rather bad in my opinion, because code with bad indentation and nested scopes is unreadable. I strongly suggest using option 3. Any comments?

Reply:
Sorry for this off-topic comment...

Reply:
On the topic of the offtopic subject. 

There is an LLVM tool made specifically for this reasons (E.g getting a standardized look on source code)

https://clang.llvm.org/docs/ClangFormat.html

I guess it should be possible to run it on the C-runtime and add it to Jenkins:)

Reply:
(I must also add I prefer how Andreas writes the brackets)

Reply:
> I would like to make a general comment on the coding style of scopes/function bodies in the c runtime:
> 
> ### Option 1
> ```
> void foo() {
>  
>     [...]
> }
> ```
> 
> ### Option 2
> ```
> void foo() {
>    [...]
> }
> ```
> 
> ### Option 3
> ```
> void foo()
> {
>     [...]
> }
> ```
> 
> This is how often we use the three patterns in the runtime:
> 
> ```
> \) \{\n\n  ->   23 hits (e.g. in line 87)
> \) \{\n    -> 2418 hits (e.g. in line 89)
> \)\n\{\n   -> 1563 hits  
> ```
> 
> So what is my point? It would be good to keep the coding consistent. It seems that option 1 isn't really used at all. Option 2 is rather bad in my opinion, because code with bad indentation and nested scopes is unreadable. I strongly suggest using option 3. Any comments?

I was all for Option 3 for years, but now I'm more in favor of Option 2 :)

Reply:
I can use option 2, no problem.
And I would like to have one code style used everywhere, but probably not enforced by Jenkins.

Reply:
I did try the clang formatter with `--style=llvm` setting (Option 2), since it is in the VSCODE C/C++ extension I'm using and I like it. Maybe I'll stick with that for now.

And maybe I must revert my previous comment on enforcing it with Jenkins. It's not a big change and uniform and easy readable code would be a really nice feature.

Reply:
> 
> 
> I did try the clang formatter with `--style=llvm` setting (Option 2), since it is in the VSCODE C/C++ extension I'm using and I like it. Maybe I'll stick with that for now.
> 
> And maybe I must revert my previous comment on enforcing it with Jenkins. It's not a big change and uniform and easy readable code would be a really nice feature.

I think we should decide the style on the most common one in the runtime (I do prefer Option 2). I have Jenkins access but little time. Maybe we should make a ticket about it and bring up for discussion